### PR TITLE
[Feature] Copy classes and instances metadata

### PR DESCRIPTION
**Pull request by shpran**
# [ADF-1030](https://oat-sa.atlassian.net/browse/ADF-1030)

## Require PR
* https://github.com/oat-sa/tao-core/pull/3526

## Companion PR
* https://github.com/oat-sa/extension-tao-backoffice/pull/144

## Tasks
- [ ] Add unit tests
- [x] Update `tao-core` dependency

### CODE REVIEW

**Review by gabrielfs7 — APPROVED** [review]
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [ ] New code is respecting code style rules
- [ ] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [ ] Acceptance criteria are respected
- [ ] Pull request title and description are meaningful

File: models/classes/Copier/ItemClassCopier.php [hunk 1]
```diff
@@ -53,10 +53,9 @@ public function copy(
 
     private function assertInItemsRootClass(core_kernel_classes_Class $class): void
     {
-        if (
-            $class->getUri() !== TaoOntology::CLASS_URI_ITEM
-            && !$class->isSubClassOf($class->getClass(TaoOntology::CLASS_URI_ITEM))
-        ) {
+        $rootClass = $class->getClass(TaoOntology::CLASS_URI_ITEM);
+
+        if (!$class->equals($rootClass) && !$class->isSubClassOf($class->getClass(TaoOntology::CLASS_URI_ITEM))) {
```
Comment (hectoras) [inline models/classes/Copier/ItemClassCopier.php:58]: I think you can avoid fetching the same class twice here

```suggestion
        $rootClass = $class->getClass(TaoOntology::CLASS_URI_ITEM);

        if (!$class->equals($rootClass) && !$class->isSubClassOf($rootClass)) {
```

**Review by hectoras — APPROVED** [review]

File: models/classes/Copier/ItemClassCopier.php (hunk 1 above)
Comment (shpran) [reply models/classes/Copier/ItemClassCopier.php:58]: Good point 🙂 
Forgot to change this, thanks!

Comment (shpran) [reply models/classes/Copier/ItemClassCopier.php]: Done

File: models/classes/Copier/ItemClassCopier.php [hunk 2]
```diff
@@ -53,10 +53,9 @@ public function copy(
 
     private function assertInItemsRootClass(core_kernel_classes_Class $class): void
     {
-        if (
-            $class->getUri() !== TaoOntology::CLASS_URI_ITEM
-            && !$class->isSubClassOf($class->getClass(TaoOntology::CLASS_URI_ITEM))
-        ) {
+        $rootClass = $class->getClass(TaoOntology::CLASS_URI_ITEM);
+
+        if (!$class->equals($rootClass) && !$class->isSubClassOf($rootClass)) {
```
Comment (gabrielfs7) [inline models/classes/Copier/ItemClassCopier.php:58]: Are tests still passing @shpran ?

**Comment by AntonNeviarouski** [comment]
QA verified

**Pull request merged**